app/models/spree/line_item_decorator.rb

```diff
@@ -127,6 +127,12 @@ def sufficient_stock?
 
   private
 
+  def update_inventory_with_scoping
+    scoper.scope(variant)
+    update_inventory_without_scoping
+  end
+  alias_method_chain :update_inventory, :scoping
+
```

Comment:
this is just example code to close the decision about "where do we do the scoping".

then we can discuss the "how do we do the scoping":
@sauloperez in this case I am not sure what 's the most elegant way of doing this. any ideas? I think that to put an hook inside spree line_item.update_inventory is almost as bad as this...